.github/workflows/e2e-tests.yml

```
@@ -74,7 +74,9 @@ jobs:
           java-version: "${{ needs.get-build-requirements.outputs.java_version || 21 }}"
 
       - name: Build uberjar with ./bin/build.sh
-        run: ./bin/build.sh
+        run: |
+          timeout 10m ./bin/build.sh
```

I think you can just do

```suggestion
        timeout-minutes: 10
        run:  ./bin/build.sh
```